File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        a = set()
        left = 0
        lmax = 0
        for right in range (len(s)):
            while s[right] in a:
                a.remove(s[left])
                left += 1
            a.add(s[right])
            lmax = max(lmax, right-left+1)
        return lmax

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int op = 0;
        ArrayList<Character> ans = new ArrayList<>();
        for(int i=0; i<s.length(); i++){
            while(ans.contains(s.charAt(i))){
                ans.remove(0);
            }
            ans.add(s.charAt(i));
            op = Math.max(op, ans.size());
        }
        return op;
    }
}